modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx

```diff
@@ -57,18 +124,32 @@ function FormPagination (props) {
     setPendingSubmission(false);
   }
 
+  let handlePageChange = (direction) => {
+    let change = (direction === "next" ? 1 : -1);
+    let nextPage = activePage;
+    while ((change === 1 || nextPage > 0) && (change === -1 || nextPage < lastValidPage())) {
+      nextPage += change;
+      if (pages[nextPage].canBeVisible) break;
+    }
+    if (nextPage !== activePage) {
+      window.scrollTo(0, 0);
+    }
+    setActivePage(nextPage);
+  }
+
   let saveButton =
     <Button
       type="submit"
       variant="contained"
       color="primary"
-      disabled={saveInProgress}
+      disabled={saveInProgress || activePage === lastValidPage()}
```

Comment:
Bug: This makes the save button always disabled on paginated forms.
```suggestion
      disabled={saveInProgress || !enableSave && activePage === lastValidPage()}
```

Reply:
Fixed in bc306fc .